crates/core/src/network/handler/lists.rs

```diff
@@ -108,16 +168,20 @@ fn create_holding_map(context: Arc<Context>) -> HashMap<EntryHash, Vec<AspectHas
 
 pub fn handle_get_gossip_list(get_list_data: GetListData, context: Arc<Context>) {
     context.clone().spawn_task(move || {
-        let authoring_map = create_authoring_map(context.clone());
+        let mut address_map = create_authoring_map(context.clone());
         let holding_map = create_holding_map(context.clone());
-
-        let address_map = authoring_map.into_iter().chain(holding_map).collect();
+        holding_map.into_iter().for_each(|(entry, set)| {
+            address_map
+                .entry(entry)
+                .or_insert_with(|| set.clone())
+                .union(&set);
+        });
```

Comment:
I don't think this does anything, because `union` does not modify a set but returns a new one! I can fix this one.